packages/block-library/src/post-featured-image/edit.js

```diff
@@ -1,35 +1,40 @@
 /**
  * WordPress dependencies
  */
-import { useEntityProp, useEntityId } from '@wordpress/core-data';
+import { useEntityProp } from '@wordpress/core-data';
 import { useSelect } from '@wordpress/data';
 import { ResponsiveWrapper } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
 
-function PostFeaturedImageDisplay() {
+function PostFeaturedImageDisplay( { context: { postId, postType } } ) {
 	const [ featuredImage ] = useEntityProp(
 		'postType',
-		'post',
-		'featured_media'
+		postType,
+		'featured_media',
+		postId
 	);
 	const media = useSelect(
 		( select ) =>
 			featuredImage && select( 'core' ).getMedia( featuredImage ),
 		[ featuredImage ]
 	);
-	return media ? (
+	if ( ! media ) {
+		return __( 'Featured Image is not set' );
+	}
+	const alt = media.alt_text || `{postType} Featured Media`;
```

Comment:
Not 100% sure about this fallback `alt`. It's definitely not applicable to `posts` though, so it should change somehow..

Reply:
I believe we use a message like "No alternative text set" ...

Reply:
I changed it :)